design/configuration-crd.md

```diff
@@ -0,0 +1,271 @@
+  cluster:
+    dns-lookup-family: auto
+  network:
+    num-trusted-hops: 0
+    admin-port: 9001
+  rateLimitService:
+    extensionService: projectcontour/ratelimit
+    domain: contour
+    failOpen: false
+    enableXRateLimitHeaders: false
+  policy:
+    request-headers:
+      set:
+    response-headers:
+      set:
+status:
+```
+
+## Contour Deployment
+
+A managed version of Contour was made available with the `Contour Operator`.
+Since Contour will manage Envoy instances, the Operator will now manage instances of Contour.
+The details of how an instance of Contour should be deployed within a cluster will be defined in the second CRD named `ContourDeployment`. 
+
+A controller will watch for these objects to be created and take action on them accordingly to make desired state in the cluster match the configuration on the spec. 
+
+```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: ContourDeployment
+metadata:
+  name: contour
+spec:
+  replicas: 2
+  nodePlacement:
+    nodeSelector:
+    tolerations:
+  configuration:
+    server:
+      xds-server-type: contour
+    ingressClassName: contour
+    envoy:
+      networkPublishing:
+      nodePlacement:
+        nodeSelector:
+        tolerations:
+    gateway:
+        controllerName: projectcontour.io/projectcontour/contour
+    disableAllowChunkedLength: false
+    disablePermitInsecure: false
+    tls:
+      minimum-protocol-version: "1.2"
+      cipher-suites:
+      - '[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]'
+      - '[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]'
+      - 'ECDHE-ECDSA-AES256-GCM-SHA384'
+      - 'ECDHE-RSA-AES256-GCM-SHA384'
+    fallback-certificate:
+      name: fallback-secret-name
+      namespace: projectcontour
+    envoy-client-certificate:
+      name: envoy-client-cert-secret-name
+      namespace: projectcontour
+    leaderelection:
+      configmap-name: leader-elect
+      configmap-namespace: projectcontour
+    enableExternalNameService: false
+    logging: 
+      accesslog-format: envoy
+      accesslog-format-string: "...\n"
+      json-fields:
+        - <Fields Omitted)
+    default-http-versions:
+      - "HTTP/2"
+      - "HTTP/1.1"
+    timeouts:
+      request-timeout: infinity
+      connection-idle-timeout: 60s
+      stream-idle-timeout: 5m
+      max-connection-duration: infinity
+      delayed-close-timeout: 1s
+      connection-shutdown-grace-period: 5s
+    cluster:
+      dns-lookup-family: auto
+    network:
+      num-trusted-hops: 0
+      admin-port: 9001
+    rateLimitService:
+      extensionService: projectcontour/ratelimit
+      domain: contour
+      failOpen: false
+      enableXRateLimitHeaders: false
+    policy:
+      request-headers:
+        set:
+      response-headers:
+        set:
+status:
+```
+
+## Processing Logic
+
+Contour will require a new flag (`--contour-config`), which will allow for customizing the name of the `ContourConfiguration` CRD that is it to use.
```

Comment:
This paragraph implies that the ContourConfiguration object can only live in the same namespace as the Contour deployment. This is fine, I think, but we should be explicit that we're not going to allow you to configure this, because cross-namespace references add security risk and shouldn't really be required here.